apps/gitness/src/pages-v2/repo/stores/branch-selector-store.ts

```diff
@@ -0,0 +1,26 @@
+import { create } from 'zustand'
+
+import { BranchSelectorListItem, IBranchSelectorStore } from '@harnessio/ui/views'
+
+export const useBranchSelectorStore = create<IBranchSelectorStore>(set => ({
+  size: 'default',
+  setSize: size => set({ size }),
+
+  selectedBranch: undefined,
+  setSelectedBranch: (selectedBranch?: BranchSelectorListItem) => set({ selectedBranch }),
+
+  branchList: undefined,
+  setBranchList: (branchList?: BranchSelectorListItem[]) => set({ branchList }),
+
+  tagList: [{ name: '', sha: '' }],
+  setTagList: (tagList: BranchSelectorListItem[]) => set({ tagList }),
+
+  repoId: '',
+  setRepoId: (repoId: string) => set({ repoId }),
+
+  spaceId: '',
+  setSpaceId: (spaceId: string) => set({ spaceId }),
+
+  prefix: '',
+  setPrefix: (prefix: string | undefined) => set({ prefix })
+}))
```

Comment:
```
prefix?: string
```

Reply:
It's an optional prop, we can skip the default value altogether, or else default it as an empty string.

Reply:
Imho we should make props optional using `?` instead of explicitly adding `undefined` as a type. Just how its done across the code base so helps maintain consistency, and is more readable.  Sorry for nitpicking 😅

Reply:
Makes sense! I too was wondering why I used undefined 😂